atlascope/core/models/dataset.py

```diff
@@ -25,7 +25,10 @@ class Dataset(TimeStampedModel, models.Model):
         default=settings.DATASET_TYPES[0],
     )
     source_dataset = models.ForeignKey(
-        'Dataset', null=True, on_delete=models.PROTECT, related_name='derived_datasets'
+        'Dataset',
+        null=True,
+        on_delete=models.PROTECT,
```

Comment:
Why `PROTECT`? If the source dataset is deleted, then it seems the `source_dataset` field of any derived datasets would become `null` (i.e., `SET_NULL` instead of `PROTECT`). What do you think?

Reply:
`PROTECT` here prevents deleting datasets that are a `source_dataset` for some derived dataset, right?. I think this is the correct behavior, especially in the case where a dataset is used across multiple investigations.

Reply:
Yes, this is saying that if B is a dataset that was derived from A, then any attempts to delete B will fail because A holds a reference to it.

I don't think this is really correct behavior. I also am not sure it matters for us in Phase 1 (since we don't have a mechanism to delete datasets, nor do we really need one right now) but in general terms, it doesn't make sense to prevent deleting a dataset just because some other dataset wants to refer to it.

In developer terms, this restriction will make it harder for us to manually set things up via the admin console. I don't think we need to change it right now (and definitely not in this PR), but if it causes problems we should revisit.